lib/chef/provider_resolver.rb

```diff
@@ -68,27 +68,39 @@ def provided_by?(provider_class)
       potential_handlers.include?(provider_class)
     end
 
-    def self.includes_handler?(resource_name, provider_class)
-      priority_map.includes_handler?(resource_name, provider_class)
-    end
-
     def enabled_handlers
-      @enabled_handlers ||=
-        potential_handlers.select { |handler| !overrode_provides?(handler) || handler.provides?(node, resource) }
+      potential_handlers.select { |handler| !overrode_provides?(handler) || handler.provides?(node, resource) }
```

Comment:
Hmmm, I typically like to do it when i can so that in later refactorings I can just re-use the method and not have to worry about performance...